### PR TITLE
Merge from wizard

### PR DESCRIPTION
**Pull request by manooweb**

### CODE REVIEW

File: modules/wizard/wizard.php [hunk 1]
```diff
@@ -0,0 +1,762 @@
+<?php
+/**
+ * Main class for Polylang wizard.
+ *
+ * @since 2.7
+ */
+class PLL_Wizard {
+	const PLUGIN_FILTER = array( 'woocommerce', 'polylang', 'polylang-pro', 'polylang-wc' );
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: Yes! I noticed it too. Its because it is used in a part of the code which is merged to PLL4WC.
This constant must be declared there.

File: modules/wizard/wizard.php [hunk 2]
```diff
@@ -0,0 +1,762 @@
...
+	}
+
+	/**
+	 * Save a activation transient when Polylang is activating to redirect to wizard
+	 *
+	 * @param string $plugin_name  plugin basename.
+	 * @param bool   $network_wide if activated for all sites in the network.
+	 * @since 2.7
+	 */
+	public function activated_plugin( $plugin_name, $network_wide ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: Yes! Exactly

[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: Fix this bug https://github.com/polylang/wizard/issues/41 and I removed the reference to PLLWC_FILE constant too. Each plugin need to be responsible of the activation of the wizard by hooking to activated_plugin by itself and using pll_activation_redirect transient.
I don't reproduce the SQL error

File: modules/wizard/view-wizard-page.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				?>
+				<?php if ( 'last' !== $this->step ) : ?>
+				<p class="pll-wizard-actions step">
+					<button
+						type="submit"
+						class="button-primary button button-large button-next"
+						value="continue"
+						name="save_step"
+					>
+						<?php esc_html_e( 'Continue', 'polylang' ); ?><span class="dashicons dashicons-arrow-right-alt2"></span>
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/view-wizard-page.php]: Yes! See the issue considering RTL 
https://github.com/polylang/wizard/issues/34

File: modules/wizard/wizard.php [hunk 4]
```diff
@@ -0,0 +1,762 @@
...
+	 *
+	 * @since 2.7
+	 */
+	public function setup_wizard_page() {
+		// if Polylang isn't activated wizard plugin can't work.
+		if ( ! defined( 'POLYLANG_VERSION' ) ) {
+			return;
+		}
+
+		if ( ! PLL_Admin_Notices::is_dismissed( 'wizard' ) ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: Yes! here https://github.com/polylang/polylang/blob/master/admin/admin-notices.php#L151
So it isn't necessary to test when we add a notice

File: modules/wizard/html-wizard-notice.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		esc_html( POLYLANG )
+	);
+	?>
+	</strong>
+	<?php
+	echo ' &#8211; ';
+	esc_html_e( 'You&lsquo;re almost ready to translate your contents!', 'polylang' );
+	?>
+</p>
+<p class="submit">
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/html-wizard-notice.php]: Keep the class but override padding and margin top.
It's interesting for sibling button in this specific paragraph.

File: modules/wizard/view-wizard-page.php [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+		<?php wp_print_styles( array_merge( $this->styles, $this->steps[ $this->step ]['styles'] ) ); ?>
+		<?php do_action( 'admin_head' ); ?>
+	</head>
+	<body class="pll-wizard wp-core-ui">
+		<h1 id="pll-logo">
+			<a href="https://polylang.pro/">
+				<img
+					src="<?php echo esc_url( plugins_url( '/modules/wizard/images/polylang-logo-back.png', POLYLANG_FILE ) ); ?>"
+					alt="Polylang"
+				/>
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/view-wizard-page.php]: Keep the img tag for reponsive reason.
Rename polylang-logo-back to polylang-logo and delete polylang-logo which is not the right version of the logo.
Remove alternative text

File: modules/wizard/wizard.php [hunk 7]
```diff
@@ -0,0 +1,762 @@
...
+
+	/**
+	 * Return html code of the wizard notice
+	 *
+	 * @since 2.7
+	 */
+	public function wizard_notice() {
+		ob_start();
+		// PLL_MODULES_INC is correctly defined relatively to the plugin folder.
+		include PLL_MODULES_INC . '/wizard/html-wizard-notice.php'; // phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.UsingCustomConstant
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: Hum! I think it was because of the use of PLL_MODULES_INC in an other plugin. Indeed now WPCS sniff doesn't say anything however WPCS Vip Minimum expects absolute path.
So I removed phpcs:ignore comment but I keep the use of absolute path with PLL_MODULES_INC constant.

File: modules/wizard/wizard.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/wizard.php]: I reproduced the SQL error. It is the first time the wizard is run when no language is set.

File: modules/wizard/view-wizard-page.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/view-wizard-page.php]: ok.
Fixes between these 2 commits
https://github.com/polylang/polylang/commit/ea12744a1d25a9d99bf84fe2ff425bc2caae9a3e
and
https://github.com/polylang/polylang/commit/892ee1ba0945e38e2b96deb986b15126cbbc0cd7

File: modules/wizard/view-wizard-page.php [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+ * Displays the wizard
+ *
+ * @since 2.7
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Don't access directly.
+};
+$admin_body_class[] = 'pll-wizard';
+$admin_body_class[] = 'wp-core-ui';
```
Comment (Chouby) [inline modules/wizard/view-wizard-page.php]: The array should be defined before adding elements.  Here:
`$admin_body_class = array( 'pll-wizard', 'wp-core-ui' );`
Note sure why PHPCS doesn't warn here.
See for ex: 3b9a648b9254ca937d452303b8379ebd3e4ae59b

File: modules/wizard/view-wizard-page.php [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+		<script type="text/javascript">
+			var ajaxurl = '<?php echo esc_url( admin_url( 'admin-ajax.php', 'relative' ) ); ?>';
+		</script>
+		<?php do_action( 'admin_enqueue_scripts' ); ?>
+		<?php wp_print_scripts( $this->steps[ $this->step ]['scripts'] ); ?>
+		<?php wp_print_styles( array_merge( $this->styles, $this->steps[ $this->step ]['styles'] ) ); ?>
+		<?php do_action( 'admin_head' ); ?>
+	</head>
+	<!-- <body class="pll-wizard wp-core-ui"> -->
+	<body class="<?php echo join( ' ', array_map( 'sanitize_key', $admin_body_class ) ); ?>">
```
Comment (Chouby) [inline modules/wizard/view-wizard-page.php]: The sanitization wasn't really necessary as all classes are hardcoded. But I guess it makes WPS happy ;-). You can remove the comment though.

File: modules/wizard/view-wizard-step-media.php [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	exit; // Don't access directly.
+};
+
+$default_options = PLL_Install::get_default_options();
+$options = wp_parse_args( get_option( 'polylang' ), $default_options );
+$media_support = $options['media_support'];
+$rtl = '';
+if ( function_exists( 'is_rtl' ) && is_rtl() ) {
+	$rtl = '-rtl';
+}
```
Comment (Chouby) [inline modules/wizard/view-wizard-step-media.php]: Why `is_rtl` would not exist?  I see none.
`$rtl = is_rtl() ? '-rtl' : '';`
should be sufficient.

Comment (manooweb) [reply modules/wizard/view-wizard-step-media.php]: function_exists( 'is_rtl' ) inspired by WP code seen somewhere.
I could have used ternary operator instead :)
Corrected

File: modules/wizard/html-wizard-notice.php [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+	?>
+	</strong>
+	<?php
+	echo ' &#8211; ';
+	esc_html_e( 'You&lsquo;re almost ready to translate your contents!', 'polylang' );
+	?>
+</p>
+<p class="buttons">
+	<a
+		href="<?php echo esc_url_raw( $wizard_url ); ?>"
```
[the remark this answers is not in the document]
Comment (manooweb) [reply modules/wizard/html-wizard-notice.php]: Corrected

File: modules/wizard/view-wizard-page.php (hunk 9 above)
Comment (manooweb) [reply modules/wizard/view-wizard-page.php]: Yes. It's for WPCS reasons: WordPress.Security.EscapeOutput.OutputNotEscaped
Indeed, now $admin_body_class isn't filterable and no external class can be added.
And I don't it will be possible in the future. At least the code will be ready here.

File: modules/wizard/view-wizard-page.php (hunk 8 above)
Comment (manooweb) [reply modules/wizard/view-wizard-page.php]: Corrected
